fastn-core/src/library2022/processor/sql.rs

```diff
@@ -7,45 +7,31 @@ pub struct DatabaseConfig {
 pub(crate) fn get_db_config() -> ftd::interpreter::Result<DatabaseConfig> {
     let db_url = std::env::var("FASTN_DB_URL").expect("FASTN_DB_URL is not set");
 
-    if db_url.contains("://") {
+    if let Some(db_url) = db_url.strip_prefix("sqlite:///") {
+        Ok(DatabaseConfig {
+            db_url: db_url.to_string(),
+            db_type: "sqlite".to_string(),
+        })
+    } else {
         let url = url::Url::parse(&db_url).expect("Invalid DB Url");
         let db_type = url.scheme().to_string();
-        let config = DatabaseConfig { db_url, db_type };
-        Ok(config)
-    } else {
-        let config = if db_url == ":memory:" {
-            DatabaseConfig {
-                db_url,
-                db_type: "sqlite".to_string(),
-            }
-        } else if let Some(path) = db_url.strip_prefix("file://") {
-            DatabaseConfig {
-                db_url: path.to_string(),
-                db_type: "sqlite".to_string(),
-            }
-        } else {
-            DatabaseConfig {
-                db_url,
-                db_type: "sqlite".to_string(),
-            }
-        };
-
-        Ok(config)
+        Ok(DatabaseConfig { db_url, db_type })
     }
 }
 
 pub async fn process(
     value: ftd::ast::VariableValue,
     kind: ftd::interpreter::Kind,
     doc: &ftd::interpreter::TDoc<'_>,
+    config: &fastn_core::Config,
 ) -> ftd::interpreter::Result<ftd::interpreter::Value> {
     let db_config = fastn_core::library2022::processor::sql::get_db_config()?;
     let db_type = db_config.db_type.as_str();
 
     match db_type {
-        "pg" => Ok(fastn_core::library2022::processor::pg::process(value, kind, doc).await?),
+        "postgres" => Ok(fastn_core::library2022::processor::pg::process(value, kind, doc).await?),
```

Comment:
Why have we changed the processor name? Also if it's needed to be changed, are we not supporting backward incompatibility?

Reply:
Actually it is not the processor name but the schema part of the connection uri eg: "postgres://". I have not removed any processor, so backward compatibility is there. people can still use those processors, they just have to use the env var `FASTN_DB_URL` instead of `FASTN_PG_URL` or `db` header for Sqlite. the `db` header has been removed as it was requested in the above mentioned PR.